// @ionic-native/network-interface/__ivy_ngcc__/ngx/index.js
import { __decorate, __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import * as ɵngcc0 from '@angular/core';
var NetworkInterface = /** @class */ (function (_super) {
    __extends(NetworkInterface, _super);
    function NetworkInterface() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    NetworkInterface.prototype.getWiFiIPAddress = function () { return cordova(this, "getWiFiIPAddress", {}, arguments); };
    NetworkInterface.prototype.getCarrierIPAddress = function () { return cordova(this, "getCarrierIPAddress", {}, arguments); };
    NetworkInterface.prototype.getHttpProxyInformation = function (url) { return cordova(this, "getHttpProxyInformation", {}, arguments); };
    NetworkInterface.pluginName = "NetworkInterface";
    NetworkInterface.plugin = "cordova-plugin-networkinterface";
    NetworkInterface.pluginRef = "networkinterface";
    NetworkInterface.repo = "https://github.com/salbahra/cordova-plugin-networkinterface";
    NetworkInterface.platforms = ["Android", "BlackBerry 10", "Browser", "iOS", "Windows", "Windows Phone"];
NetworkInterface.ɵfac = function NetworkInterface_Factory(t) { return ɵNetworkInterface_BaseFactory(t || NetworkInterface); };
NetworkInterface.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: NetworkInterface, factory: function (t) { return NetworkInterface.ɵfac(t); } });
var ɵNetworkInterface_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(NetworkInterface);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NetworkInterface, [{
        type: Injectable
    }], null, null); })();
    return NetworkInterface;
}(IonicNativePlugin));
export { NetworkInterface };
